### PR TITLE
[CELEBORN-18] Refactor stream manager to distinguish map partition and reduce partition.

### PR DESCRIPTION
**Pull request by FMX**
… reduce partition.

# [CELEBORN-18] Refactor stream manager to distinguish map partition and reduce partition.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java [hunk 1]
```diff
@@ -28,24 +28,39 @@
...
+import org.apache.celeborn.common.protocol.PartitionType;
 import org.apache.celeborn.common.util.Utils;
 
 public class FileInfo {
   private final String filePath;
   private final List<Long> chunkOffsets;
   private final UserIdentifier userIdentifier;
+  private final PartitionType partitionType;
 
   public FileInfo(String filePath, List<Long> chunkOffsets, UserIdentifier userIdentifier) {
```
Comment (RexXiong) [inline common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:40]: suggest use this(filePath....)

File: common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java [hunk 2]
```diff
@@ -28,24 +28,39 @@
...
+      List<Long> chunkOffsets,
+      UserIdentifier userIdentifier,
+      PartitionType partitionType) {
+    this.filePath = filePath;
+    this.chunkOffsets = chunkOffsets;
+    this.userIdentifier = userIdentifier;
+    this.partitionType = partitionType;
+  }
+
+  public FileInfo(String filePath, UserIdentifier userIdentifier, PartitionType partitionType) {
```
Comment (RexXiong) [inline common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:58]: ditto

Comment (FMX) [reply common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:55]: done

File: common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java [hunk 3]
```diff
@@ -28,32 +28,53 @@
...
+    this(
+        filePath,
+        new ArrayList() {
+          {
+            add(0L);
+          }
+        },
+        userIdentifier,
+        PartitionType.REDUCE);
```
Comment (waitinfuture) [inline common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:64]: partitionType

File: common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java [hunk 4]
```diff
@@ -28,32 +28,53 @@
...
 
-  public FileInfo(String filePath, UserIdentifier userIdentifier) {
-    this.filePath = filePath;
-    this.chunkOffsets = new ArrayList<>();
-    chunkOffsets.add(0L);
-    this.userIdentifier = userIdentifier;
+  public FileInfo(String filePath, UserIdentifier userIdentifier, PartitionType partitionType) {
+    this(
+        filePath,
+        new ArrayList() {
```
Comment (waitinfuture) [inline common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:58]: Better to use one line for the new ArrayList

File: common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java [hunk 5]
```diff
@@ -28,32 +28,53 @@
...
 
   @VisibleForTesting
   public FileInfo(File file, UserIdentifier userIdentifier) {
-    this.filePath = file.getAbsolutePath();
-    this.chunkOffsets = new ArrayList<>();
-    chunkOffsets.add(0L);
-    this.userIdentifier = userIdentifier;
+    this(
+        file.getAbsolutePath(),
+        new ArrayList() {
```
Comment (waitinfuture) [inline common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:71]: ditto

**Review by waitinfuture — APPROVED** [review]
LGTM

**Pull request merged**